### DISCUSSION TITLE
OpenEVSE with battery

### DISCUSSION THREAD

**Discussion by Dylan811806** [OpenEVSE/open_evse#26] [Q&A] [answered]
Hello,
I'm working on a project using OpenEVSE and a battery, basically, I have an issue for a special case of power loss.
When I lose the power for the T2 plug, my board is still powered by the battery, and I don't detect anything for this case.
If I'm in a session, and I have a power loss, the Atmega detect it, cut the session and send an error "GroundFailure" to the ESP, but the error is "erase" since the session is ended, so the last status I got is "Available". 
And if I'm not in a session, it didn't detect it at all. I measured the value on the optocoupler (AC_TEST and AC_TEST_2) (the one that manage the test for the relay) and the value that is sent to the Atmega don't change in both case.
Someone have an idea to detect this kind of power loss without making any hardware modification ?
Thanks

**Comment by chris1howell** [comment, marked as answer]
The current OpenEVSE hardware monitors ground and stuck relay on two
channels one on each AC line on the LOAD side of the relay. If the relay is
closed it monitors ground if the relay is open the same chip monitor for a
stick relay.

UL has recently changed the requirements for ground monitoring from "while
charging" to "continuously". A new hardware revision is coming that
combines each line with diodes. One channel monitors ground continuously
and the other monitors stuck relay an the load side of the relay. The
changes comply with new requirements while sacrificing a non-safety feature
of auto L1/L2 detection by using the 2 channels differently.

The current hardware would require significant modification to monitor
ground on the line side of the relay and monitor stuck relay on the load
side. It would be possible to move the test leads from the load side to
line side however an essential safety feature (stuck relay detection) would
have to be disabled.

**Reply by DylanVautrin** [reply]
Ok, thanks for support
